mxcubeweb/routes/ra.py

```diff
@@ -43,25 +43,27 @@ def handle_timeout_gives_control(sid, timeout=30):
                 return make_response(msg, 409)
 
         current_user.requests_control = True
-        server.user_datastore.commit()
+        app.usermanager.update_user(current_user)
+
+        server.emit("observersChanged", namespace="/hwr")
 
         gevent.spawn(
             handle_timeout_gives_control,
             current_user.username,
             timeout=10,
         )
 
-        app.usermanager.emit_observers_changed()
-
         return make_response("", 200)
 
     @bp.route("/cancel_request", methods=["POST"])
     @server.restrict
     def cancel_request():
         """Cancel request for control"""
         current_user.requests_control = False
-        server.user_datastore.commit()
-        app.usermanager.emit_observers_changed()
+        app.usermanager.update_user(current_user)
+
+        server.emit("observersChanged", namespace="/hwr")
```

Comment:
Same, the user state is now refetched right after calling the endpoint in the front-end, in the `cancelControlRequest` thunk.